### PR TITLE
fix(python): Parse tuples in array accesses

### LINKED ISSUE #3387: [Autofix] sometimes drops closing bracket in python slices
**Describe the bug**
The closing bracket is sometimes dropped in the autofix.

**To Reproduce**
https://semgrep.dev/s/juliusdegesys:closingbracketbugrepro
```yaml
rules:
- id: test
  message: Bug Repro
  severity: INFO
  pattern: f($X)
  fix: f($X, 1)
```
Input file:
```python
# These are ok:
f(x[0])
f([0, 1])
f(x(0, 1))
# Closing square bracket of $X gets dropped in autofix below:
f(x[0, 1])
f(x[0, 1, 2])
f(x[0, 1:2])
f(x['a', 'b'])
```
Output:
```
severity:info rule:test: Bug Repro
autofix: f(x[0], 1)
2:f(x[0])
--------------------------------------------------------------------------------
autofix: f([0, 1], 1)
3:f([0, 1])
--------------------------------------------------------------------------------
autofix: f(x(0, 1), 1)
4:f(x(0, 1))
--------------------------------------------------------------------------------
autofix: f(x[0, 1, 1)
6:f(x[0, 1])
--------------------------------------------------------------------------------
autofix: f(x[0, 1, 2, 1)
7:f(x[0, 1, 2])
--------------------------------------------------------------------------------
autofix: f(x[0, 1:2, 1)
8:f(x[0, 1:2])
--------------------------------------------------------------------------------
autofix: f(x['a', 'b', 1)
9:f(x['a', 'b'])
ran 1 rules on 1 files: 7 findings
```

**Expected behavior**
Closing bracket not to be dropped in autofix.
```
severity:info rule:test: Bug Repro
autofix: f(x[0], 1)
2:f(x[0])
--------------------------------------------------------------------------------
autofix: f([0, 1], 1)
3:f([0, 1])
--------------------------------------------------------------------------------
autofix: f(x(0, 1), 1)
4:f(x(0, 1))
--------------------------------------------------------------------------------
autofix: f(x[0, 1], 1)
6:f(x[0, 1])
--------------------------------------------------------------------------------
autofix: f(x[0, 1, 2], 1)
7:f(x[0, 1, 2])
--------------------------------------------------------------------------------
autofix: f(x[0, 1:2], 1)
8:f(x[0, 1:2])
--------------------------------------------------------------------------------
autofix: f(x['a', 'b'], 1)
9:f(x['a', 'b'])
ran 1 rules on 1 files: 7 findings
```

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [X] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

**Environment**
If not using semgrep.dev: are you running off docker, an official binary, a local build?
```
pip install semgrep
$ semgrep --version
0.56.0
```

### PR DESCRIPTION
**Pull request by nmote**
This fixes both a matching issue and an autofix issue (see tests).

Fixes #3387

Test plan: Automated tests

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: cli/tests/e2e/snapshots/test_autofix/test_autofix/rulesautofixpython-ranges.yaml-autofixpython-ranges.py-dryrun/results.json [hunk 1]
```diff
@@ -211,6 +211,170 @@
         "line": 5,
         "offset": 49
       }
+    },
```
Comment (aryx) [inline cli/tests/e2e/snapshots/test_autofix/test_autofix/rulesautofixpython-ranges.yaml-autofixpython-ranges.py-dryrun/results.json:214]: We should get rid of those results.json. Does someone read them and make sure
they contain the right thing?
Seems like what matters is that the generated code look good (and there's a .py-fixed nice file for that.
The results.json file seems redundant and unreadable anyway).

**Review by aryx — APPROVED** [review]

File: cli/tests/e2e/snapshots/test_autofix/test_autofix/rulesautofixpython-ranges.yaml-autofixpython-ranges.py-dryrun/results.json (hunk 1 above)
Comment (nmote) [reply cli/tests/e2e/snapshots/test_autofix/test_autofix/rulesautofixpython-ranges.yaml-autofixpython-ranges.py-dryrun/results.json:214]: I wrote the test and updated the snapshot before applying the fix, and then looked at the diff in the `results.json` file after applying the fix, but I agree that its utility is dubious especially when with autofix you can see exactly what the result of the autofix will be.

**Comment by nmote** [comment]
I still have to figure out the failing benchmark test

**Comment by nmote** [comment]
I updated the benchmark snapshots with the new findings. It looks like they are legitimate findings that were missed before. I added a new semgrep-core test with a reduced version of one.

**Comment by emjin** [comment]
Nice! Love to see it!

**Pull request merged**